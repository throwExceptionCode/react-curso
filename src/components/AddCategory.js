import React, { useState } from 'react';
import PropTypes from 'prop-types';


const AddCategory = ({setCategories}) => {
    const [inputValue, setinputValue] = useState('');

    const handleInputCahnge = (e)=>{
        setinputValue(e.target.value);
    };
    const handleSubmit = (e)=>{
        e.preventDefault();
        if (inputValue.trim().length > 2 ) {
            setCategories(cats => [inputValue, ...cats] );
            setinputValue('');
        }
        
        
    };
    return (
        <form onSubmit={ handleSubmit } >
            <h2>AddCategory</h2>
                <input 
                    type='text'
                    value={ inputValue }
                    onChange={ handleInputCahnge }
                />
            </form>
    )
}

AddCategory.propTypes = {
    setCategories : PropTypes.func.isRequired,
};

export default AddCategory

