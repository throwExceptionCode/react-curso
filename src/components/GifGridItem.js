import React from 'react';
import PropTypes from 'prop-types';

export const GifGridItem = ( { title, url, id } ) => {
    return (
        <div className='card animate__animated animate__backInDown' >
            <h5 className='TituloImagen'>{ title }</h5>
            <img src={url} alt={title}/>
            <h5>{ id }</h5>
        </div>
    )
}

GifGridItem.propTypes = {
    title : PropTypes.string.isRequired,
    id : PropTypes.string.isRequired,
    url: PropTypes.string.isRequired,
};