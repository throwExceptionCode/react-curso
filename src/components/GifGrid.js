import React from 'react';//, { useState,useEffect }
import { useFetchGifs } from '../hooks/useFechGifts';
//import { getGif } from '../helpers/getGifs';
import { GifGridItem } from './GifGridItem';

const GifGrid = ({ category }) => {
    //const [Images, setImages] = useState([]);
    const { data:images,loading } = useFetchGifs(category);
    //const [conteo, setconteo] = useState(1);
    /*useEffect(()=>{
        getGif(category).then(setImages);
    },[category])*/
    //<h2>{conteo}</h2>
    //<button onClick={ () => setconteo( conteo + 1 ) } >+</button>
    
    
    
    //getGif

    useFetchGifs();
    
    return (
        <>
            <li>
                <div className="card-grid" >
                    <h3  className="title-search" > { category } </h3>
                    {loading && 'Cargando' }
                    <br/>
                    { 
                        images.map( (img) => (
                            <GifGridItem key={img.id}  { ...img } />
                        ) )
                    }
                </div>
            </li>

                
        </>
    )
}

export default GifGrid
