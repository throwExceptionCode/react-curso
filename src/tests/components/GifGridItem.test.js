
//import { render } from "@testing-library/react";

import React from 'react';
import { shallow } from "enzyme";
import { GifGridItem } from '../../components/GifGridItem';

describe('Pruebas GifGridItem', () => {
    const title = 'Titulo';
    const url = 'https://localhost/img.jpg';
    let wrapper = shallow(<GifGridItem title={ title } url={ url } id={ 1 } />);
    beforeEach(()=>{
        wrapper = shallow(<GifGridItem title={ title } url={ url } id={ 1 } />);
    });

    
    test('should be match with snapshot', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('should be p like title', () => {
        const h5 = wrapper.find('.TituloImagen').text().trim();
        expect(h5).toBe(title); 
    });
    test('should be img from url and alt of props', () => {
        const img = wrapper.find('img').props();
        expect(img.src).toBe(url); 
        expect(img.alt).toBe(title); 
    })
    
    test('should be animated css', () => {
        const div = wrapper.find('div').props();
        //expect(div.className.search('animate__backInDown')>1 && true).toBe(true);
        expect(div.className.includes('animate__backInDown')).toBe(true);
    })
    
    

    
    
    
});

