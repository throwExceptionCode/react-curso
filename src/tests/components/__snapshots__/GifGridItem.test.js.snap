// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas GifGridItem should be match with snapshot 1`] = `
<div
  className="card animate__animated animate__backInDown"
>
  <h5
    className="TituloImagen"
  >
    Titulo
  </h5>
  <img
    alt="Titulo"
    src="https://localhost/img.jpg"
  />
  <h5>
    1
  </h5>
</div>
`;
